Modules/Sources/JetpackStats/Extensions/Calendar+Navigation.swift

```diff
@@ -0,0 +1,111 @@
+import Foundation
+
+extension Calendar {
+    enum NavigationDirection {
+        case backward
+        case forward
+
+        var systemImage: String {
+            switch self {
+            case .backward: "chevron.backward"
+            case .forward: "chevron.forward"
+            }
+        }
+    }
+
+    /// Navigates to the next or previous period from the given date interval.
+    ///
+    /// This method navigates by the length of the period for the given component.
+    /// For example, if the interval represents 6 months and the component is `.month`,
+    /// it will navigate forward or backward by 6 months.
+    ///
+    /// - Parameters:
+    ///   - interval: The date interval to navigate from
+    ///   - direction: Whether to navigate forward (.forward) or backward (.backward)
+    ///   - component: The calendar component to use for navigation
+    /// - Returns: A new date interval representing the navigated period
+    func navigate(_ interval: DateInterval, direction: NavigationDirection, component: Calendar.Component) -> DateInterval {
```

Comment:
I don't get the meaning of the `component` parameter.

Take one of the unit tests as an example:

```
// 7-day period should navigate by 7 days
(Date("2025-01-08T00:00:00-03:00"), Date("2025-01-15T00:00:00-03:00"), Calendar.Component.day, ...)

// Using above test arguments makes call to
navigate(
  DateInterval(start: "2025-01-08", end: "2025-01-15"),
  direction: .backwards,
  component: .day
)
```

Why passing `.day` as the `component` argument? Is the `component` argument even necessary?

The `navigate` function calculates the immediate previous or next period of a given date interval. It feels like only the first two arguments are needed.

Reply:
I just saw the `determineNavigationComponent` function below. Is the `component` argument here always `determineNavigationComponent(interval)` in practice? If that's the case, we can omit the `component` and call `determineNavigationComponent` internally within the `navigate` function?